traits/tests/test_configure_traits.py

```diff
@@ -126,3 +127,31 @@ def modify_model(*args, **kwargs):
 
         self.assertIsInstance(unpickled, Model)
         self.assertEqual(unpickled.count, model.count)
+
+    def test_edit_when_false(self):
+        # Check for deprecation warning when *edit* is false.
+        model = Model()
+        with mock.patch.object(self.toolkit, "view_application") as mock_view:
+            with self.assertWarns(DeprecationWarning):
+                model.configure_traits(edit=False)
+        mock_view.assert_not_called()
+
+    def test_edit_when_true(self):
+        # Check for deprecation warning when *edit* is false.
+        model = Model()
+        with mock.patch.object(self.toolkit, "view_application") as mock_view:
+            mock_view.return_value = True
+            with self.assertWarns(DeprecationWarning):
+                model.configure_traits(edit=True)
+        self.assertEqual(len(mock_view.mock_calls), 1)
```

Comment:
It would be clearer to use `mock_view.assert_called_once()` here. We couldn't do that before, since we were keeping compatibility with Python 3.5 (`assert_called_once` is new in Python 3.6).

I'll update.

Reply:
Done.